oshi-core/src/main/java/oshi/util/platform/windows/WmiUtil.java

```diff
@@ -137,20 +137,24 @@ public static boolean hasNamespace(String namespace) {
         } catch (COMException e) {
             // Ignore any exceptions with OpenHardwareMonitor
             if (!OHM_NAMESPACE.equals(query.getNameSpace())) {
-                switch (e.getHresult().intValue()) {
-                case Wbemcli.WBEM_E_INVALID_NAMESPACE:
-                    LOG.warn("COM exception: Invalid Namespace {}", query.getNameSpace());
-                    break;
-                case Wbemcli.WBEM_E_INVALID_CLASS:
-                    LOG.warn("COM exception: Invalid Class {}", query.getWmiClassName());
-                    break;
-                case Wbemcli.WBEM_E_INVALID_QUERY:
-                    LOG.warn("COM exception: Invalid Query: {}", queryToString(query));
-                    break;
-                default:
-                    LOG.warn(
-                            "COM exception querying {}, which might not be on your system. Will not attempt to query it again. Error was: {}:",
-                            query.getWmiClassName(), e.getMessage());
+                if (e.getHresult() == null) {
```

Comment:
I think it might be better to create a temp int variable and assign 0 to it if `getHresult()` returns null.  Then just use this integer in the switch, and you'll get the existing longer/verbose default log message.

Reply:
Could we consider to use this code `throw new COMException("Failed to initialize COM library.", new HRESULT(0));` instead ? If so, the correction in only 2 lines in WmiUtil and what I have submited is not needed anymore.

Reply:
Yes, except use the existing hres variable (not sure why I missed that before.  I think I had to exclude it with the 4.x version of JNA and missed adding it when moving to 5.x (although 5.x created the default null that bit us here.)